admin/class-aioseop-notices.php

```diff
@@ -0,0 +1,714 @@
+		 *
+		 * @since 3.0
+		 *
+		 * @uses AIOSEOP_PLUGIN_DIR . 'admin/display/notice-default.php' Template for default notices.
+		 *
+		 * @return void
+		 */
+		public function display_notice_default() {
+			$this->display_notice( 'default' );
+		}
+
+		/**
+		 * Display Notice as AIOSEOP Screens
+		 *
+		 * Method for Admin notices exclusive to AIOSEOP screens.
+		 * NOTE: As of 3.0, display_notice_default() & display_notice_aioseop()
+		 * have the same functionality, but serves as a future development concept.
+		 *
+		 * @since 3.0
+		 *
+		 * @uses AIOSEOP_PLUGIN_DIR . 'admin/display/notice-aioseop.php' Template for notices.
+		 *
+		 * @return void
+		 */
+		public function display_notice_aioseop() {
+			$this->display_notice( 'aioseop' );
+		}
+
+		/**
+		 * Display Notice
+		 *
+		 * @since 2.8
+		 *
+		 * @param string $template Slug name for template.
+		 */
+		public function display_notice( $template ) {
+			if ( ! wp_script_is( 'aioseop-admin-notice-js', 'enqueued' ) || ! wp_style_is( 'aioseop-admin-notice-css', 'enqueued' ) ) {
+				return;
+			} elseif ( 'default' !== $template && 'aioseop' !== $template ) {
+				return;
+			} elseif ( ! current_user_can( 'aiosp_manage_seo' ) ) {
+				return;
+			}
+
+			$current_screen  = get_current_screen();
+			$current_user_id = get_current_user_id();
+			foreach ( $this->active_notices as $a_notice_slug => $a_notice_time_display ) {
+				$notice_show = true;
+
+				// Screen Restriction.
+				if ( ! empty( $this->notices[ $a_notice_slug ]['screens'] ) ) {
+					// Checks if on aioseop screen.
+					if ( in_array( 'aioseop', $this->notices[ $a_notice_slug ]['screens'], true ) ) {
+						if ( ! in_array( $current_screen->id, $this->aioseop_screens, true ) ) {
+							continue;
+						}
+					}
+
+					// Checks the other screen restrictions by slug/id.
+					if ( ! in_array( 'aioseop', $this->notices[ $a_notice_slug ]['screens'], true ) ) {
+						if ( ! in_array( $current_screen->id, $this->notices[ $a_notice_slug ]['screens'], true ) ) {
+							continue;
+						}
+					}
+				}
+
+				// User Settings.
+				if ( 'user' === $this->notices[ $a_notice_slug ]['target'] ) {
+					$user_dismissed = get_user_meta( $current_user_id, 'aioseop_notice_dismissed_' . $a_notice_slug, true );
+					if ( ! $user_dismissed ) {
+						$user_notice_time_display = get_user_meta( $current_user_id, 'aioseop_notice_display_time_' . $a_notice_slug, true );
+						if ( ! empty( $user_notice_time_display ) ) {
+							$a_notice_time_display = intval( $user_notice_time_display );
+						}
+					} else {
+						$notice_show = false;
+					}
+				}
+
+				// Display/Render.
+				if ( defined( 'DISABLE_NAG_NOTICES' ) && true === DISABLE_NAG_NOTICES && ( 'notice-error' !== $this->notices[ $a_notice_slug ]['class'] || 'notice-warning' !== $this->notices[ $a_notice_slug ]['class'] || 'notice-do-nag' !== $this->notices[ $a_notice_slug ]['class'] ) ) {
```

Comment:
! in_array( $this->notices[ $a_notice_slug ]['class'], array( 'notice-error', 'notice-warning', 'notice-do-nag'), true ) might be better?

Reply:
Changed. That would be much better.